### PR TITLE
Inference Server App - HIP support

### PR DESCRIPTION
**Pull request by hansely**
Added HIP support for inference server app
Modified the cmake to build and install inference_server_app under /opt/rocm/bin

### CODE REVIEW

File: apps/cloud_inference/server_app/CMakeLists.txt [hunk 1]
```diff
@@ -52,6 +47,7 @@ list(APPEND SOURCES
 		configure.cpp
 		compiler.cpp
 		inference.cpp
+    	inference_hip.cpp
```
Comment (rrawther) [inline apps/cloud_inference/server_app/CMakeLists.txt:50]: please adjust the spacing here

File: apps/cloud_inference/server_app/CMakeLists.txt [hunk 2]
```diff
@@ -30,15 +30,10 @@ set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/../../../amd_openvx/cmake)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 
-find_package(OpenCL QUIET)
```
Comment (rrawther) [inline apps/cloud_inference/server_app/CMakeLists.txt:35]: Need to do OpenCV version related changes for supporting both versions(3&4) of OpenCV

File: apps/cloud_inference/server_app/inference.cpp [hunk 3]
```diff
@@ -5,7 +5,7 @@
 #include <chrono>
 #include <dlfcn.h>
 #include <opencv2/opencv.hpp>
-#include <highgui.h>
+#include <opencv2/highgui.hpp>
```
Comment (rrawther) [inline apps/cloud_inference/server_app/inference.cpp:8]: I think this needs to be changed for OpenCV 4 and above.

File: apps/cloud_inference/server_app/CMakeLists.txt [hunk 4]
```diff
@@ -81,5 +119,5 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
 	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
 	set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /MTd")
 else()
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -msse4.1 -mf16c")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -msse4.1 -mf16c")
```
Comment (rrawther) [inline apps/cloud_inference/server_app/CMakeLists.txt:122]: please add -fopenmp flag as well.

**Review by rrawther — CHANGES_REQUESTED** [review]
Please update based on comments

**Comment by kiritigowda** [comment]
@hansely use this for OpenCV support in your CMakeList - https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/49eac66a6a828bcf8d2c0021da7fc6769c66a345/samples/inference/mv_objdetect/CMakeLists.txt#L65

**Comment by hansely** [comment]
@kiritigowda Ready to be merged

File: apps/cloud_inference/server_app/CMakeLists.txt [hunk 5]
```diff
@@ -61,25 +57,73 @@ list(APPEND SOURCES
...
+	    target_compile_definitions(inference_server_app PUBLIC USE_OPENCV_4=1)
+    else()
+	    target_compile_definitions(inference_server_app PUBLIC USE_OPENCV_4=0)
+    endif()
 	include_directories(${OpenCV_INCLUDE_DIRS})
 	target_link_libraries(inference_server_app ${OpenCV_LIBRARIES})
-else(OpenCV_FOUND)
+else()
+	message(FATAL_ERROR "OpenCV -- Version-${OpenCV_VERSION_MAJOR}.${OpenCV_VERSION_MINOR}.X Not Supported")
 	target_compile_definitions(inference_server_app PUBLIC ENABLE_OPENCV=0)
```
[the remark this answers is not in the document]
Comment (kiritigowda) [reply apps/cloud_inference/server_app/CMakeLists.txt:121]: @hansely take a look at this comment

[the remark this answers is not in the document]
Comment (hansely) [reply apps/cloud_inference/server_app/CMakeLists.txt]: @kiritigowda removed the definition part

**Review by kiritigowda — APPROVED** [review]

**Pull request merged**